src/main/kotlin/com/atlassian/performance/tools/jvmtasks/api/TaskTimer.kt

```diff
@@ -17,12 +18,34 @@ object TaskTimer {
         task: () -> T
     ): T {
         CloseableThreadContext.push(label).use {
+            eventHandlers.forEach { it.onTaskStarted(label) }
             val start = now()
-            val result = task()
+            val result =
+                try {
+                    task()
+                } catch (e: Exception) {
+                    val duration = Duration.between(start, now())
+                    eventHandlers.forEach { it.onTaskFailed(label, duration) }
+                    throw e
+                }
             val duration = Duration.between(start, now())
-
+            eventHandlers.forEach { it.onTaskSucceeded(label, duration) }
             logger.debug("The task took $duration")
             return result
         }
     }
+
+    fun registerEventHandler(handler: EventHandler) {
+        eventHandlers.add(handler)
+    }
+
+    fun resetEventHandlers() {
+        eventHandlers.clear()
+    }
+
+    interface EventHandler {
```

Comment:
The `TaskTimer` singleton is `@ThreadSafe`, so its event handlers need to be like that too.
```suggestion
    @ThreadSafe
    interface EventHandler {
```